src/api/source/mvt.js

```diff
@@ -0,0 +1,175 @@
+import * as rsys from '../../client/rsys';
+import Dataframe from '../../core/dataframe';
+import * as Protobuf from 'pbf';
+import { VectorTile } from '@mapbox/vector-tile';
+import { decodeLines, decodePolygons } from '../../client/mvt/feature-decoder';
+import TileClient from './TileClient';
+import Base from './base';
+
+const geometryTypes = {
+    UNKNOWN: 'unknown',
+    POINT: 'point',
+    LINE: 'line',
+    POLYGON: 'polygon'
+};
+
+export default class MVT extends Base {
+
+    /**
+     * Create a carto.source.MVT.
+     *
+     * @param {object} data - A MVT data object
+     * @param {object} metadata - A carto.source.mvt.Metadata object
+     *
+     * @example
+     * const metadata = new carto.source.mvt.Metadata([{ type: 'number', name: 'total_pop'}])
+     * new carto.source.MVT("https://{server}/{z}/{x}/{y}.mvt", metadata);
+     *
+     * @fires CartoError
+     *
+     * @constructor MVT
+     * @extends carto.source.Base
+     * @memberof carto.source
+     * @api
+     */
+    // FIXME, multi subdomain
+    constructor(templateURL, metadata = { columns: [] }) {
+        super();
+        this._templateURL = templateURL;
+        this._metadata = metadata;
+        this._tileClient = new TileClient(templateURL);
+        this._categoryStringToIDMap = {};
+        this._numCategories = 0;
+    }
+
+    _clone() {
+        return new MVT(this._templateURL, JSON.parse(JSON.stringify(this._metadata)));
```

Comment:
Why do we need `JSON.parse(JSON.stringify(this._metadata))`?

Reply:
I wanted to play it safe, a clone should not share state with its "cloner"

Reply:
should we create an `util.clone` method?

Reply:
The first question for a method like that will be? Is it shallow? Is it deep? Will it allow circular dependencies? With the current implementation, you don't need to look up those in our internal utils since it only use the built-in utils.